src/arkode/arkode_arkstep.c

```diff
@@ -66,15 +65,6 @@ void* ARKStepCreate(ARKRhsFn fe, ARKRhsFn fi, sunrealtype t0, N_Vector y0,
     return (NULL);
   }
 
-  /* Test if all required vector operations are implemented */
-  nvectorOK = arkStep_CheckNVector(y0);
-  if (!nvectorOK)
-  {
-    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
-                    MSG_ARK_BAD_NVECTOR);
-    return (NULL);
-  }
```

Comment:
[suggestion] To fail faster, maybe call `arkCheckNvector` here (and similarly in the other constructors) instead of waiting until `arkInit` is called at the end of this function

Reply:
I agree it would be preferable to have this check in one place. Since `arkInit` is at the end `arkAllocVec` should check that `clone` is defined since the constructors call it before the operation checks happen